File: java-code/src/Solns/maximum_depth_of_binary_tree_104.java
package Solns;

public class maximum_depth_of_binary_tree_104 {



	public int maxDepth(TreeNode root) {
		// If the current node is null, return 0 since it won't contribute to the max depth 
		if(root == null)
            return 0;
        
		// Otherwise traverse left and then right nodes 		
		int left = maxDepth(root.left);
		int right = maxDepth(root.right);
		// Take the max of the results of the left and right recursive calls and add 1
		return Math.max(left, right) + 1;
	}

	public static void main(String[] args) {
		
	}

}

File: python-code/maximum_depth_of_binary_tree_104.py
class TreeNode:
  def __init__(self, val):
    self.val = val
    self.left, self.right = None, None



def maxDepth(root):
    if(root == None):
        return 0

    return 1 + max(maxDepth(root.left), maxDepth(root.right))

def main():
    root = TreeNode(12)
    root.left = TreeNode(7)
    root.right = TreeNode(1)
    root.left.left = TreeNode(9)
    root.right.left = TreeNode(10)
    root.right.right = TreeNode(5)
    print(maxDepth(root))


main()

